### ISSUE TITLE
LDAP auth works only with sugarcrm not with suitecrm

### ISSUE THREAD

**Issue by tomys** [salesagility/SuiteCRM-Outlook-Plugin#23]
if i try to connect to a new installed suitecrm 7.7.x or to a suitecrm installation which was migrated vom sugarcrm the ldap authetication of the outlook addin does not work. -> "Authetication failed".

If i try to connect to our sugarcrm system with the suitcrm outlook plugin the login works fine!

The "normal" login (without ldap) works fine on both systems sugarcrm an suitecrm.

**Comment by tomys** [comment]
I found out that the outlook addin sends two time the password. The first time it sends the md5 and then clear text password. Thats strange but it should not be a problem.
The problem is, that both passwords (md5 and text) are includes three "^@" characters at the end.

So i have modified the < LDAPAuthenticateUser.php > to remove these special escape characters.

If the outlook plugin was fixed this code can be removed from the suitecrm LDAPAuthenticateUser.php .

// gto NEW -->
                    $password = str_replace("^@" , "", $password);
 // <-- NEW gto
                      // MRF - Bug #18578 - punctuation was being passed as HTML entities, i.e. &amp;
                $bind_password = html_entity_decode($password,ENT_QUOTES);
...

Why sugarcrm can handle the password with the special characters at the end should also be researched.

Regards,
Thomas
